tests/types/test_enum.py

```diff
@@ -0,0 +1,125 @@
+from enum import Enum
+
+import pytest
+from psycopg.types.enum import EnumInfo, register_enum
+
+from psycopg import pq
+
+
+class _TestEnum(str, Enum):
```

Comment:
Why the underscore?

Reply:
pytest shows error if class name starts with `Test`:
```
E   pytest.PytestCollectionWarning: cannot collect test class 'TestEnum' because it has a __new__ constructor (from: tests/types/test_enum.py)
```
Maybe rename it (`StrTestEnum`)?

Reply:
Ah you are right, Test is collected. Yes, something like StrTestEnum, MyEnum, MyStrEnum would do.

Reply:
Rewritten tests in more universal way, to use different enums, labels, formats and encodings. Can you check it please?